### PR TITLE
navigating-across-documents: revisions and explanations for 001, 002, 003 & 004

### PR DESCRIPTION
**Pull request by rwaldron**
The results of manually running these tests:

Test | Chrome Canary | Edge | Firefox Nightly | Safari Technology Preview
-- | -- | -- | -- | --
001.html | ✔ | ✘ 001-1.html's unload handler is not aborted | ✔ | ✔
002.html | ✔ | ✔ | ✔ | ✔
003.html | ✔ | ✘ 003-2.html's unload handler is not aborted | ✘ 003-2.html's unload handler is not aborted | ✔
004.html | ✔ | ✘ 004-2.html's unload handler is not aborted | ✔ | ✔

Notes: 

- Edge is consistently misbehaving, issue filed here: https://developer.microsoft.com/en-us/microsoft-edge/platform/issues/14320545/
- Firefox only misbehaves for only a single case, issue filed here: https://bugzilla.mozilla.org/show_bug.cgi?id=1410515

### CODE REVIEW

**Comment by w3c-bots** [comment]
# Build PENDING

Started: None
Finished: None





View more information about this build on:

- [WPT PR Status](http://pulls.web-platform-tests.org/build/19770)
- [TravisCI](https://travis-ci.org/w3c/web-platform-tests/builds/290617228)

**Comment by annevk** [comment]
The cleanup looks good. Feel free to use const/let btw, if you want.

I'm a little concerned about the list of steps in a comment. I strongly suspect they will get out of sync as folks modify the test over time without taking a holistic view of the situation. Perhaps brief clarifying comments intertwined with the code would do better. @bzbarsky @domenic?

**Comment by domenic** [comment]
+1 for commenting nearby the code if possible.

Why make these stylistic changes? Adding comments is good but I don't see a motivation for changing the style, variable names, etc. It just messes up the blame for subjective gain. Was this discussed somewhere else perhaps?

In a related vein, are the "revisions" normative or only stylistic? I can't really tell given all the stylistic changes.

Awesome to see the issues filed and the tests run in general.

**Comment by rwaldron** [comment]
> I'm a little concerned about the list of steps in a comment.

Originally I tried keeping the commenting inline, but the outcome was subjectively hard to follow; I switched to a "walkthrough" style of documenting the operations and (again: subjectively) found it easier to track across all of the files I was working on. For example, as you navigate from test file to test file, having the step-by-step explanations in the same place and same in the same form makes it easier to follow and understand the bigger coverage picture.

> Why make these stylistic changes?

These files had many different code styles, which was distracting (and in some cases confusing or even misleading) when working through documenting and determining points of spec coverage and relevance. As part of my efforts, I determined that it was in the best interest of the test material for it to all be uniform.

File: html/browsers/browsing-the-web/navigating-across-documents/001.html [hunk 1]
```diff
@@ -1,20 +1,31 @@
...
 <script src="/resources/testharnessreport.js"></script>
 <div id="log"></div>
 <iframe src="001-1.html"></iframe>
 <script>
-var t = async_test();
-onload = t.step_func(function() {
-  var iframe = document.getElementsByTagName("iframe")[0];
-  var new_src = iframe.src.replace(/\d{3}-\d\.html/, "001-2.html");
-  iframe.src = new_src;
+var test = async_test();
```
Comment (Ms2ger) [inline html/browsers/browsing-the-web/navigating-across-documents/001.html]: Please move as much of the code as possible into the test steps, for example:

```js
async_test(function() {
  var iframe = document.querySelector("iframe");
  window.onload = this.step_func(function() {
    iframe.src = iframe.src.replace(/\d{3}-\d\.html/, "001-2.html");
  });
  window.onmessage = this.step_func_done(function(e) {
    assert_equals(e.data, "001-2");
  });
});
```

**Comment by rwaldron** [comment]
@bzbarsky _nudge_

**Comment by bzbarsky** [comment]
I'm really sorry for the lag here... I keep ending up trying to review this when I'm too tired to make sure I'm not missing something important.  :(

I will try to deal with this first thing Monday.

**Comment by bzbarsky** [comment]
I am not going to get to this review.  Please ask a different reviewer.  :(
